<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office">

<head>
<link rel="stylesheet" href="style.css" type="text/css" />
<meta http-equiv="Content-Language" content="en-us" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<script type="text/javascript">
var sc_project=7569123;
var sc_invisible=1;
var sc_security="404f14b2";
</script>
<script type="text/javascript"
src="http://www.statcounter.com/counter/counter.js"></script>
<noscript><div class="statcounter"><a title="vBulletin
statistic" href="http://statcounter.com/vbulletin/"
target="_blank"><img class="statcounter"
src="http://c.statcounter.com/7569123/0/404f14b2/1/"
alt="vBulletin statistic"></a></div></noscript>
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-27596315-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>
<title>User Guide</title>
</head>

<body>
<div id="content">
	<div id="main">
		<div id="myTitle">
		<h1>rMATS v4.0.2 (turbo)</h1>
		<h2><a href="http://www.mimg.ucla.edu/faculty/xing/" target="_blank">Xing Lab</a>, University of California, Los Angeles</h2>
		</div>
		<div id="top">
			<ul id="nav">
				<li><a href="index.html">Home</a></li>
				<li style="color:white;font-size:22px;font-weight:bold;">|</li>
				<li><a style="text-decoration: underline;" href="download.html">Downloads</a></li>
				<li style="color:white;font-size:22px;font-weight:bold;">|</li>
				<li><a href="user_guide.htm">Documents</a></li>
				<li style="color:white;font-size:22px;font-weight:bold;">|</li>
				<li><a href="faq.html">FAQ</a></li>
				<li style="color:white;font-size:22px;font-weight:bold;">|</li>
				<li><a href="http://www.mimg.ucla.edu/faculty/xing/" target="_blank">Xing Lab</a></li>
			</ul>
		</div>
		<div style="background-color: lightgray;">
		<div id="right">
			<ul class="reference">
				<li><a href="#version" class="reference">Which version to use</a><br></li>
				<li><a href="#install" class="reference">Installation</a><br></li>
				<li><a href="#test" class="reference">Test rMATS v4.0.2 (turbo) on PC3E & GS689 dataset:</a><br></li>
				<li><a href="#using" class="reference">Using rMATS v4.0.2 (turbo)</a><br></li>
				<li><a href="#output" class="reference">Output</a><br></li>
				<li><a href="#as_events" class="reference">Alternative Splicing Events</a><br></li>
			</ul>

			<h2 id="version">Which version to use:</h2>
			<div id="subSection">
				rMATS v4.0.2 (turbo) was built with two different settings of Python interpreter. In order to know which version you should use, you need to check what Unicode type your Python is built with. Open python console and type in:<br>
				<div class="code">>>> import sys<br>>>> print  sys.maxunicode<br>1114111</div>
				This output indicates that your python is built with --enable-unicode=ucs4, and you should use rMATS-turbo-xxx-UCS4.
				<div class="code">>>> import sys<br>>>> print  sys.maxunicode<br>65535</div>
				This output indicates that your python is built with --enable-unicode=ucs2, and you should use rMATS-turbo-xxx-UCS2.
			</div>

			<h2 id="install">Installation:</h2>
			<div id="topBotSpace">
				<table id="bulletPoint"><tr><td><ul><li>Install
				<a target="_blank" href="http://www.python.org/download/releases/2.7.3/">Python 2.7.x</a> and corresponding versions of
				<a target="_blank" href="http://www.scipy.org/Download">NumPy</a>, <a target="_blank" href="http://www.netlib.org/blas/">BLAS</a>, <a target="_blank" href="http://www.netlib.org/lapack/">LAPACK</a>, <a target="_blank" href="https://www.gnu.org/software/gsl/">GSL</a> and <a target="_blank" href="https://gcc.gnu.org/wiki/GFortran">gfortran (Fortran 77 library needed)</a>.
				<br>
				At the time of compiling this software, the default C++ compiler on Mac OS X doesn't fully support OpenMP. In order to enjoy multi-threading feature on Mac OS X, <a href="https://www.gnu.org/software/gcc/gcc-5/">GCC 5</a> is needed.
				The following is the installation procedure we used in testing:</li></ul></td></tr></table>
				<div id="subSection"><div class="code" style="height:240px">
					# For CentOS 6:<br>
					pip install numpy<br>
					yum install lapack-devel blas-devel<br>
					yum install gsl-devel.x86_64<br>
					yum install gcc-gfortran<br>
					<br>
					# For Ubuntu 14:<br>
					pip install numpy<br>
					sudo apt-get install libblas-dev liblapack-dev<br>
					sudo apt-get install libgsl0ldbl<br>
					sudo apt-get install gfortran<br>
					<br>
					# For Mac OS X Yosemite 10.10.5 (Using Homebrew for package management):<br>
					brew install gcc@5<br>
					brew install gsl<br>
					pip install numpy<br>
				</div>
				<table id="bulletPoint"><tr><td><ul><li>Download and install <a href="https://github.com/alexdobin/STAR" target="_blank">STAR </a>(version 2.5 or later) for FASTQ input.</li></ul></td></tr></table>
				<table id="bulletPoint"><tr><td><ul><li>Download <a href="rMATS.4.0.2.tgz">rMATS v4.0.2 (turbo)</a></li></ul></td></tr></table>
				<table id="bulletPoint"><tr><td><ul><li>Obtain STAR genome index for genome by either of the following two ways
					<table id="bulletPoint"><tr><td><ul><li>Download <a href="http://labshare.cshl.edu/shares/gingeraslab/www-data/dobin/STAR/STARgenomes/">pre-built STAR indexes</a> if using Human (hg38, hg19) or Mouse (mm10). </li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li>Build your own STAR index following <a href="https://github.com/alexdobin/STAR/tree/master/doc" target="_blank">STAR manual</a> from genome fasta sequence</ul></td></tr></table>
				</li></ul></td></tr></table>
				<table id="bulletPoint"><tr><td><ul><li>Untar rMATS and STAR indexes. For example, assuming you use pre-build STAR genome indexes, unpack rMATS.4.0.2.tgz in your working directory and unpack STARindex.tgz in your working directory:</a></li></ul></td></tr></table>
                <div id="subSection"><div class="code" style="height:120px">tar -xzf rMATS.4.0.2.tgz<br>cd rMATS.4.0.2/<br>... # move/copy/download data to this folder.<br>tar -xzf gtf.tgz<br>tar -xzf testData.tar</div></div>
			</div>

			<h2 id="test">Test rMATS v4.0.2 (turbo) on PC3E & GS689 dataset:</h2>
			<div id="subSection">
				Run rmats.py as below to test rMATS runs properly.<br>
				<div class="code">cd rMATS.4.0.2/<br>python rMATS-turbo-xxx-UCSx/rmats.py --b1 b1.txt --b2 b2.txt --gtf gtf/Homo_sapiens.Ensembl.GRCh37.75.gtf --od bam_test -t paired --readLength 101 --cstat 0.0001 --libType fr-unstranded</div>
                Output can be found in the bamTest directories. The test run output should look like the <a href="user_guide.htm#output">rMATS output description</a>.<br>
			</div>

			<h2 id="using">Using rMATS v4.0.2 (turbo):</h2>
			<div id="subSection">
				The following is a detailed description of the options used with rMATS v4.0.2 (turbo).<br/></br>
				<strong>Usage:</strong><br/>
					<div id="subSection">
						Running with <strong>fastq</strong>
						<div class="demo">python rmats.py --s1 s1.txt --s2 s2.txt --gtf gtfFile --bi STARindexFolder -od outDir -t readType -readLength readLength [options]*</div><br>
						Running with <strong>bam</strong>
						<div class="demo">python rmats.py --b1 b1.txt --b2 b2.txt --gtf gtfFile --od outDir -t readType --nthread nthread --readLength readLength --tstat tstat [options]*</div>
					</div>
				<strong>Required Parameters:</strong>
				<div id="subSection" style="border-left: 5px solid lightgray; padding-left:10px;">
					<table class="invisTable">
					<tr><td style="width:275px;"><strong>--s1</strong> s1.txt</td><td> A text file contains FASTQ file(s) for the sample_1.(<strong>Only if using fastq</strong>)</td></tr>
					<tr><td><strong>--s2</strong> s2.txt</td><td> A text file contains FASTQ file(s) for the sample_2.(<strong>Only if using fastq</strong>)</td></tr>
					<tr><td><strong>--b1</strong> b1.txt</td><td>A text file records mapping results for the sample_1 in bam format. (<strong>Only if using bam</strong>)</td></tr>
					<tr><td><strong>--b2</strong> b2.txt</td><td>A text file records mapping results for the sample_2 in bam format. (<strong>Only if using bam</strong>)</td></tr>
					<tr><td><strong>-t</strong> readType</td><td>Type of read used in the analysis. readType is either 'paired' or 'single'. 'paired' is for paired-end data and 'single' is for single-end data</td></tr>
					<tr><td><strong>--readLength</strong> &ltint&gt</td><td>The length of each read</td></tr>
					<tr><td><strong>--gtf</strong> gtfFile</td><td> An annotation of genes and transcripts in GTF format</td></tr>
					<tr><td><strong>--bi</strong> STARIndexFolder </td><td>The folder name of the STAR binary indexes (i.e., the name of the folder that contains SA file). For example, use ~/STARindex/hg19 for hg19. (<strong>Only if using fastq</strong>)</td></tr>
					<tr><td><strong>--od</strong> outDir </td><td>The output directory</td></tr>
					</table>
				</div>

				<strong>Optional:</strong>
				<div id="subSection" style="border-left: 5px solid lightgray; padding-left:10px;">
					<table class="invisTable">
					<tr><td style="width:275px;"><strong>--tophatAnchor</strong> &LTint&GT</td><td>The "anchor length" or "overhang length" used in the aligner. At least “anchor length” NT must be mapped to each end of a given junction. The default is 6. (<strong>This parameter applies only if using fastq</strong>)</td></tr>
					<!-- <tr><td><strong>-r1</strong> &LTfloat&GT[,&LTfloat&GT]*</td><td>The insert sizes of sample_1 data. This applies only for the paired-end data. Replicates are separated by comma. The default is 100 for each replicate</td></tr>
					<tr><td><strong>-r2</strong> &LTfloat&GT[,&LTfloat&GT]*</td><td>The insert sizes of sample_2 data. This applies only for the paired-end data. Replicates are separated by comma. The default is 100 for each replicate</td></tr>
					<tr><td><strong>-sd1</strong> &LTfloat&GT[,&LTfloat&GT]*</td><td>The standard deviations for the r1. Replicates are separated by comma. The default is 300 for each replicate</td></tr>
					<tr><td><strong>-sd2</strong> &LTfloat&GT[,&LTfloat&GT]*</td><td>The standard deviations for the r2. Replicates are separated by comma. The default is 300 for each replicate</td></tr>
					 -->
					 <tr><td><strong>--nthread</strong> &LTfloat&GT</td><td>The number of thread. The optimal number of thread should be equal to the number of CPU core.</td></tr>
					 <tr><td><strong>--cstat</strong> &LTfloat&GT</td><td>The cutoff splicing difference. The cutoff used in the null hypothesis test for differential splicing. The default is 0.0001 for 0.01% difference. Valid: 0 &#8804 cutoff &LT 1</td></tr>
					 <tr><td><strong>--tstat</strong> &LTfloat&GT</td><td>The number of thread for statistical model.</td></tr>
					 <tr><td><strong>--statoff</strong></td><td> Turn statistics part off. </td></tr>
					<tr><td><strong>-libType</strong> libraryType</td><td>Library type. Default is unstranded (<strong>fr-unstranded</strong>). Use <strong>fr-firststrand</strong> or <strong>fr-secondstrand</strong> for strand-specific data.</td></tr>
                    </table>
				</div>
				<strong>Examples:</strong>
				<div id="subSection">
					Example using <strong>fastq</strong>.
					<div class="code" style="height:80px">$cat s1.txt:<br>231ESRP.25K.rep-1.R1.fastq:231ESRP.25K.rep-1.R2.fastq,231ESRP.25K.rep-2.R1.fastq:231ESRP.25K.rep-2.R2.fastq<br>$cat s2.txt:<br>231EV.25K.rep-1.R1.fastq:231EV.25K.rep-1.R2.fastq,231EV.25K.rep-2.R1.fastq:231EV.25K.rep-2.R2.fastq<br>$</div><br>
					<div class="code">
					python rMATS-turbo-xxx-UCSx/rmats.py --s1 s1.txt --s2 s2.txt --gtf gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf --bi ~/STARindex/hg19  --od out_test -t paired --nthread 6 --readLength 101 --tophatAnchor 8 --cstat 0.0001 --tstat 6</div></br>
					Example using <strong>bam</strong>.
					<div class="code" style="height:80px">$cat b1.txt:<br>231ESRP.25K.rep-1.bam,231ESRP.25K.rep-2.bam
						<br>$cat b2.txt:<br>231EV.25K.rep-1.bam,231EV.25K.rep-2.bam
						<br>$
					</div>
					<br>
					<div class="code">python rMATS-turbo-xxx-UCSx/rmats.py --b1 b1.txt --b2 b2.txt -gtf gtf/Homo_sapiens.Ensembl.GRCh37.75.gtf -od bam_test -t paired --readLength 101 --cstat 0.0001 --libType fr-unstranded</div>
				</div>
			</div>

			<h2 id="output">Output:</h2>
			<div id="subSection">
				All output files are in --od which contains rMATS output of AS events, all possible alternative splicing (AS) events derived from GTF and RNA
				<table id="bulletPoint"><tr><td><ul><li><strong>AS_Event.MATS.JC.txt</strong> evaluates splicing with only reads that span splicing junctions
					<table id="bulletPoint"><tr><td><ul><li><i>IJC_SAMPLE_1:</i> inclusion junction counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SJC_SAMPLE_1:</i> skipping junction counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IJC_SAMPLE_2:</i> inclusion junction counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SJC_SAMPLE_2:</i> skipping junction counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
				</li></ul></td></tr></table>
				<table id="bulletPoint"><tr><td><ul><li><strong>AS_Event.MATS.JCEC.txt</strong> evaluates splicing with reads that span splicing junctions and reads on target (striped regions on <a href="splicing.jpg">home page figure</a>)
					<table id="bulletPoint"><tr><td><ul><li><i>IC_SAMPLE_1:</i> inclusion counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SC_SAMPLE_1:</i> skipping counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IC_SAMPLE_2:</i> inclusion counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SC_SAMPLE_2:</i> skipping counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
				</li></ul></td></tr></table>
				<table id="bulletPoint"><tr><td><ul><li>Important columns contained in output files above.
					<table id="bulletPoint"><tr><td><ul><li><i>IncFormLen:</i> length of inclusion form, used for normalization</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SkipFormLen:</i> length of skipping form, used for normalization</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IncLevel1:</i> inclusion level for SAMPLE_1 replicates (comma separated) calculated from normalized counts</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IncLevel2:</i> inclusion level for SAMPLE_2 replicates (comma separated) calculated from normalized counts</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IncLevelDifference:</i> average(IncLevel1) - average(IncLevel2)</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>P-Value:</i> Significance of splicing difference between two sample groups. <strong>(Only available if statistical model is on)</strong></li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>FDR:</i> False Discovery Rate calculated from p-value. <strong>(Only available if statistical model is on)</strong></li></ul></td></tr></table>
				</li></ul></td></tr></table>
				<table id="bulletPoint"><tr><td><ul><li><strong>fromGTF.AS_Event.txt</strong> all possible alternative splicing (AS) events derived from GTF and RNA.
					<table id="bulletPoint"><tr><td><ul><li><i>IJC_SAMPLE_1:</i> inclusion junction counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SJC_SAMPLE_1:</i> skipping junction counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IJC_SAMPLE_2:</i> inclusion junction counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SJC_SAMPLE_2:</i> skipping junction counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IncFormLen:</i> length of inclusion form, used for normalization</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SkipFormLen:</i> length of skipping form, used for normalization</li></ul></td></tr></table>
				</li></ul></td></tr></table>

				<table id="bulletPoint"><tr><td><ul><li><strong>JC.raw.input.AS_Event.txt</strong> evaluates splicing with only reads that span splicing junctions
					<table id="bulletPoint"><tr><td><ul><li><i>IJC_SAMPLE_1:</i> inclusion junction counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SJC_SAMPLE_1:</i> skipping junction counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IJC_SAMPLE_2:</i> inclusion junction counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SJC_SAMPLE_2:</i> skipping junction counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IncFormLen:</i> length of inclusion form, used for normalization</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SkipFormLen:</i> length of skipping form, used for normalization</li></ul></td></tr></table>
				</li></ul></td></tr></table>
				<table id="bulletPoint"><tr><td><ul><li><strong>JCEC.raw.input.AS_Event.txt</strong> evaluates splicing with reads that span splicing junctions and reads on target (striped regions on <a href="splicing.jpg">home page figure</a>)
					<table id="bulletPoint"><tr><td><ul><li><i>IC_SAMPLE_1:</i> inclusion counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SC_SAMPLE_1:</i> skipping counts for SAMPLE_1, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IC_SAMPLE_2:</i> inclusion counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SC_SAMPLE_2:</i> skipping counts for SAMPLE_2, replicates are separated by comma</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>IncFormLen:</i> length of inclusion form, used for normalization</li></ul></td></tr></table>
					<table id="bulletPoint"><tr><td><ul><li><i>SkipFormLen:</i> length of skipping form, used for normalization</li></ul></td></tr></table>
				</li></ul></td></tr></table>
			</div>

			<h2 id="as_events">Alternative Splicing Events</h2>
			<div id="subSection">
				rMATS analyzes skipped exon (SE), alternative 5' splice site (A5SS), alternative 3' splice site (A3SS), mutually exclusive exons (MXE), and retained intron (RI) events. Possible alternative splicing
				events are identified from the RNA-Seq data and annotation of transcripts in GTF format. The following is a list of provided GTF files:
				<table id="bulletPoint"><tr><td><ul><li><strong>Human</strong>, <i>Homo sapiens</i> (Ensembl or UCSC Known Genes)</li></ul></td></tr></table>
				<p id="gtf">Alternatively, you can download your own transcript annotation in GTF format. However, the first column (chromosome/contig name) in the GTF <strong>must match</strong> the sequence names in your STARindex.
			</div>

		</div>
		<div style="clear:both;"></div>
		</div>
		<div id="bot"></div>
	</div>
</div>


</body>

</html>
